packages/medusa/src/services/claim.ts

```diff
@@ -204,6 +209,102 @@ export default class ClaimService extends TransactionBaseService {
     )
   }
 
+  protected async validateCreateClaimInput(
+    data: CreateClaimInput
+  ): Promise<void> {
+    const lineItemServiceTx = this.lineItemService_.withTransaction(
+      this.manager_
+    )
+
+    const { type, claim_items, additional_items, refund_amount } = data
+
+    for (const item of claim_items) {
+      const line = await lineItemServiceTx.retrieve(item.item_id, {
+        relations: ["order", "swap", "claim_order", "tax_lines"],
+      })
+
+      if (
+        line.order?.canceled_at ||
+        line.swap?.canceled_at ||
+        line.claim_order?.canceled_at
+      ) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Cannot create a claim on a canceled item.`
+        )
+      }
+    }
+
+    if (type !== ClaimType.REFUND && type !== ClaimType.REPLACE) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Claim type must be one of "refund" or "replace".`
+      )
+    }
+
+    if (type === ClaimType.REPLACE && !additional_items?.length) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Claims with type "replace" must have at least one additional item.`
+      )
+    }
+
+    if (!claim_items?.length) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Claims must have at least one claim item.`
+      )
+    }
+
+    if (refund_amount && type !== ClaimType.REFUND) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Claim has type "${type}" but must be type "refund" to have a refund_amount.`
+      )
+    }
+  }
+
+  protected findClaimLinesOnOrder(
+    order: Order,
+    claimItems: CreateClaimItemInput[]
+  ) {
+    return claimItems.map((ci) => {
+      const allOrderItems = order.items
+
+      if (order.swaps?.length) {
```

Comment:
Suggestion: 

const additionalItems = [...(order.swaps || []), ...(order.claims || [])]

additilnalItems.forEach(...)

Reply:
> const additionalItems = [...(order.swaps || []), ...(order.claims || [])]

We have to iterate the items for each swap and claim, so this wont work

Reply:
Oups yes indeed miss read, but the if including the constraint will. It will avoid to re loop all the items to check the id match the ci.item_id though

Reply:
i ve checked your commit and it is good to me, after my comment above we should be good to merge

Reply:
Yes, sorry you are right! I'll add that one in 💪

Reply:
So orderItems = order.items.filter and then the other ones .push

Reply:
As there can only be a single item where the id matches, wouldn't it be better to return the item early?
// map order.items -> return if found
// map swap.items -> return if found
// map claim.items -> return if found